src/workerd/jsg/iterator.h

```diff
@@ -445,6 +451,92 @@ class AsyncGenerator final {
   friend class GeneratorWrapper;
 };
 
+template <typename T>
+Promise<kj::Maybe<T>> AsyncGenerator<T>::next(Lock& js) {
+  KJ_IF_SOME(i, impl) {
+    KJ_SWITCH_ONEOF(i->state) {
+      KJ_CASE_ONEOF(finsihed, typename Impl::Finished) {
```

Comment:
Nit
```suggestion
      KJ_CASE_ONEOF(finished, typename Impl::Finished) {
```
Or maybe just `_`.